lib/src/main/java/com/diffplug/spotless/FormatterProperties.java

```diff
@@ -0,0 +1,303 @@
+				Properties properties = new Properties();
+				InputStream inputProperties = new FileInputStream(file);
+				try {
+					properties.load(inputProperties);
+				} finally {
+					inputProperties.close();
+				}
+				return properties;
+			}
+		},
+
+		XML("xml") {
+			@Override
+			protected Properties execute(final File file) throws IOException, IllegalArgumentException {
+				Node rootNode = getRootNode(file);
+				String nodeName = rootNode.getNodeName();
+				if (null == nodeName) {
+					throw new IllegalArgumentException("XML document does not contain a root node.");
+				}
+				return XmlParser.parse(file, rootNode);
+			}
+
+			private Node getRootNode(final File file) throws IOException, IllegalArgumentException {
+				try {
+					/*
+					 * The parser does not validate, since the root node is only
+					 * used to decide on further processing.
+					 */
+					DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+					DocumentBuilder db = dbf.newDocumentBuilder();
+					return db.parse(file).getDocumentElement();
+				} catch (SAXException | ParserConfigurationException e) {
+					throw new IllegalArgumentException("File has no valid XML syntax.", e);
+				}
+
+			}
+
+		};
+		private static final char FILE_EXTENSION_SEPARATOR = '.';
+
+		private final List<String> supportedFileNameExtensions;
+
+		private FileParser(final String... supportedFileNameExtensions) {
+			this.supportedFileNameExtensions = Arrays.asList(supportedFileNameExtensions);
+		}
+
+		protected abstract Properties execute(File file) throws IOException, IllegalArgumentException;
+
+		public static Properties parse(final File file) throws IOException, IllegalArgumentException {
+			String fileNameExtension = getFileNameExtension(file);
+			for (FileParser parser : FileParser.values()) {
+				if (parser.supportedFileNameExtensions.contains(fileNameExtension)) {
+					return parser.execute(file);
+				}
+			}
+			String msg = String.format(
+					"The file name extension '%1$s' is not part of the supported file extensions [%2$s].",
+					fileNameExtension, Arrays.toString(FileParser.values()));
+			throw new IllegalArgumentException(msg);
+
+		}
+
+		private static String getFileNameExtension(File file) {
+			final String fileName = file.getName();
+			String fileNameExt = "";
+			int seperatorPos = fileName.lastIndexOf(FILE_EXTENSION_SEPARATOR);
+			if (0 < seperatorPos) {
+				fileNameExt = fileName.substring(seperatorPos + 1);
+			}
+			return fileNameExt;
+		}
+
+	}
+
+	private enum XmlParser {
+		PROPERTIES("properties") {
+			@Override
+			protected Properties execute(final File xmlFile, final Node rootNode)
+					throws IOException, IllegalArgumentException {
+				final Properties properties = new Properties();
+				InputStream xmlInput = new FileInputStream(xmlFile);
+				try {
+					properties.loadFromXML(xmlInput);
+				} finally {
+					xmlInput.close();
+				}
```

Comment:
:thumbsup:
Sorry, before these change I only worked with Java 6. Thanks for all the tips ad hints...